### PR TITLE
Add bookmark table to std.typecons

### PR DESCRIPTION
**Pull request by wilzbach**
Found a weird-looking bookmark table in std.functional and then had a look at other bookmark tables and found that std.typecons was in deep need:

Before:

image

After:

image

Other interesting modules
- [std.process](https://dlang.org/phobos/std_process.html)
- [std.variant](https://dlang.org/phobos/std_variant.html)
- [std.uri](https://dlang.org/phobos/std_uri.html)
- [std.random](https://dlang.org/phobos/std_random.html)
- [std.parallelism](https://dlang.org/phobos/std_parallelism.html)
- [std.numeric](https://dlang.org/phobos/std_numeric.html)
- [std.mathspecial](https://dlang.org/phobos/std_mathspecial.html)
- [std.getopt](https://dlang.org/phobos/std_getopt.html)

### CODE REVIEW

File: std/functional.d [hunk 1]
```diff
@@ -23,7 +23,7 @@ $(TR $(TH Function Name) $(TH Description)
     $(TR $(TD $(LREF forward))
         $(TD Forwards function arguments while saving ref-ness.
     ))
-    $(TR $(TD $(LREF lessThan)), $(LREF greaterThan)), $(D $(LREF equalTo)
+    $(TR $(TD $(LREF lessThan), $(LREF greaterThan), $(D $(LREF equalTo))
```
Comment (andralex) [inline std/functional.d]: So this is a bug fix? Nice.

Comment (wilzbach) [inline std/functional.d]: Yes, that's why it's based against stable & I realized that `std.typecons` is need of a bookmark table ;-)

**Comment by MetaLang** [comment]
What's this SHA mismatch message?

**Comment by wilzbach** [comment]
> What's this SHA mismatch message?

https://github.com/dlang/phobos/pull/5625#issuecomment-315921337

File: std/functional.d [hunk 2]
```diff
@@ -23,7 +23,7 @@ $(TR $(TH Function Name) $(TH Description)
     $(TR $(TD $(LREF forward))
         $(TD Forwards function arguments while saving ref-ness.
     ))
-    $(TR $(TD $(LREF lessThan)), $(LREF greaterThan)), $(D $(LREF equalTo)
+    $(TR $(TD $(LREF lessThan), $(LREF greaterThan), $(LREF equalTo))
```
Comment (MetaLang) [inline std/functional.d]: Shouldn't this be a separate PR?

Comment (JackStouffer) [inline std/functional.d]: IIRC there were some issues in the past about formatting in relation to this

**Comment by wilzbach** [comment]
Splitted of in https://github.com/dlang/phobos/pull/5641 and https://github.com/dlang/phobos/pull/5642 as GH sends wrong hooks after a rebase (they have been notified and "are working on it")...
